src/main/java/uk/nhs/cactus/common/audit/AuditParser.java

```diff
@@ -0,0 +1,30 @@
+package uk.nhs.cactus.common.audit;
+
+import static java.util.function.Predicate.not;
+import static org.apache.commons.lang3.StringUtils.strip;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuditParser {
+  public Map<String, Collection<String>> getHeadersFrom(String headers) {
+    return Stream.of(headers.split("\n"))
+        .map(header -> header.split(":"))
```

Comment:
This split should probably have a limit set since you only want the first colon to be split

Reply:
Good spot, done!